docs/src/interface.md

```diff
@@ -11,11 +11,109 @@ Depth = 2
 
+We would like to highlight a few of the types and functions in the next two sections before listing the remaining types and functions alphabetically.
+
+### The Manifold Type
+
+Besides the most central type, that of a [`AbstractManifold`](@ref) accompanied by [`AbstractManifoldPoint`](@ref) to represent points thereon, note that the point type is meant in a lazy fashion.
+This is mean as follows: if you implement a new manifold and your points are represented by matrices, vectors or arrays, then it is best to not type the poitns `p` in functions, such that the methods work for example for other array representation types as well.
+You should subtype your new points on a manifold, if the structure you use is more structured, see for example[`FixedRankMatrices`](@ref).
+Another reason is, if you want to distinguish (and hence dispatch on) different representation of points on the manifold.
+For an example, see the [Hyperbolic](@ref HyperbolicSpace) manifold, which has different models to be represented.
+
+```@autodocs
+Modules = [Manifolds, ManifoldsBase]
+Pages = ["maintypes.jl"]
+Order = [:type, :function]
+```
+
+### The exponential and the logarithmic map and geodesics
+
+Geodesics are the generalizations of a straight line to manifolds, i.e. their intrinsic acceleration is zero.
+Together with geodesics one also obtains the exponential map and its inverse, the logarithmic map.
+Informally speaking, the exponential map takes a vector (think of a direction and a length) at one point and returns another point,
+which lies towards this direction at distance of the specified length. The logarithmic map does the inverse, i..e. given two points, what is the vector that “points towards” the other point.
+
+```@autodocs
+Modules = [Manifolds, ManifoldsBase]
+Pages = ["exp_log_geo.jl"]
+Order = [:function]
+```
+
+### Retractions and inverse Retractions
+
+The exponential and logarithmic map might be too expensive to evaluate or not be available in a very stable numerical way. Retractions provide a possibly cheap, fast and stable alternative.
+
+The following figure compares the exponential map [`exp`](@ref)`(M, p, X)` on the [`Circle`](@ref)`(ℂ)` (or [`Sphere`](@ref)`(1)` embedded in $ℝ^2$ with one possible retraction, the one based on projections. Note especially that ``\mathrm{dist}(p,q)=\lVert X\rVert_p`` while this is not the case for ``q'``.
+
+![A comparson of the exponential map and a retraction on the Circle.](assets/images/retraction_illustration_600.png)
+
+```@autodocs
+Modules = [Manifolds, ManifoldsBase]
+Pages = ["retractions.jl"]
+Order = [:function]
+```
+
+To distinguish different types of retractions, the last argument of the (inverse) retraction
+specifies a type. The following ones are available.
+
+```@autodocs
+Modules = [Manifolds, ManifoldsBase]
+Pages = ["retractions.jl"]
+Order = [:type]
+```
+
+### Projections
```

Comment:
It may also make sense to illustrate projections on Rotations (even 2D ones) since it's noticeably different.

Reply:
Can you elaborate a little on this? At first glance I don‘t see how that‘s different – and also I would not want to clutter this too much. It is illustrating one difference (project tangent and project point), so maybe the rotation projection could also be done over on the rotations tutorial or on the rotations manifold page?

Reply:
Yes, maybe that's not the best place to explain fine details. The reason I wrote about Rotations is that tangent vectors to rotations are represented in the Lie algebra at identity. Currently rotation to that tangent space (i.e. representation change) is not performed, so I don't think tangent vector projection for rotations is the pushforward of point projections.

Reply:
oh, if so, a change to the Lie group presentation should be performed, yes (if that has a different representation) but sure there it is a little different, also the parallel transport is (due to the representation, and just the representation) the identity.

Reply:
The issue is already tracked in https://github.com/JuliaManifolds/Manifolds.jl/issues/360 so we can leave it for another PR.

Reply:
Perfect, cool.